pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -669,33 +694,24 @@ func (d *deployer) deploy(ctx context.Context, operation string) (extensionsv1al
 	case extensionsv1alpha1.OperatingSystemConfigPurposeProvision:
 		if features.DefaultFeatureGate.Enabled(features.UseGardenerNodeAgent) {
 			units, files = initUnits, initFiles
+
+			// Add gardener-user and sshd-ensurer when SSH access for the node is enabled
+			if d.sshAccessEnabled {
+				for _, c := range []components.Component{gardeneruser.New(), sshdensurer.New()} {
+					cUnits, cFiles, err := c.Config(componentsContext)
+					if err != nil {
+						return nil, err
+					}
+					units = append(units, cUnits...)
```

Comment:
In this case we will have the `gardener-user` and `sshd-ensurer` added twice to the original OSC components, as the init OSC components are also added to them ref: https://github.com/gardener/gardener/blob/c29cc0fe30c09b980cb81d4a1caea6b7bccad244/pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go#L709
and
https://github.com/gardener/gardener/blob/c29cc0fe30c09b980cb81d4a1caea6b7bccad244/pkg/component/extensions/operatingsystemconfig/original/original.go#L59-L79

Could this cause any problems?

Reply:
[`gardener-user` and `sshd-ensurer`](https://github.com/gardener/gardener/blob/483a626aaa4d4aef83a1c3e83ef15a9ab163627d/pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go#L700-L707) are no part of [`initUnits`](https://github.com/gardener/gardener/blob/483a626aaa4d4aef83a1c3e83ef15a9ab163627d/pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go#L672-L676). Thus, they shouldn't be added twice.

Afaik there is no validation which prevents the same unit from being added multiple times. The `node-agent` would merge the units in the end. This would not look nice, but we would end up with the same unit anyway.
However, I think the units are added to original OSC only once.